hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/KeepAliveCache.java

```diff
@@ -156,11 +181,11 @@ synchronized void evictIdleConnection() {
    *
    * @param hc HttpClientConnection to be closed
    */
-  private void closeHtpClientConnection(final HttpClientConnection hc) {
+  private void closeHttpClientConnection(final HttpClientConnection hc) {
     try {
       hc.close();
-    } catch (IOException ignored) {
-
+    } catch (IOException ex) {
+      LOG.debug("Close failed for connection: " + hc, ex);
```

Comment:
use ` for connection: {}", hc, ex)`, so hc.toString() is only invoked if logging at debug *and* if the method raised an exception, that it is caught and swallowed.

Reply:
Good point, have taken it. Also, in KeepAliveCache and AbfsConnectionManager, before doing a LOG.debug, have added a check for LOG.isDebugEnabled().